name_variant/examples/test.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use name_variant::NamedVariant;
+
+#[derive(NamedVariant)]
+enum TestEnum {
+    A,
+    B(),
+    C(i32, i32),
+    D { _name: String, _age: i32 },
+    VariantTest,
+}
+
+fn main() {
+    let x = TestEnum::C(1, 2);
+    assert_eq!(x.variant_name(), "C");
+
+    let x = TestEnum::A;
+    assert_eq!(x.variant_name(), "A");
+
+    let x = TestEnum::B();
+    assert_eq!(x.variant_name(), "B");
+
+    let x = TestEnum::D {
+        _name: "Jane Doe".into(),
+        _age: 30,
+    };
+    assert_eq!(x.variant_name(), "D");
```

Comment:
Could we add something that uses x after the assert so we can be sure that variant_name() does not move x and allows x to be used later?

Reply:
I'm now forcing the receiver, using `(&x).variant_name()` syntax.